test/util.cpp

```diff
@@ -740,6 +740,58 @@ TEST(util, double_to_int_limits)
+
+    EXPECT_TRUE(std::numeric_limits<uint8_t>::min() == double_to_int<uint8_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<uint16_t>::min() == double_to_int<uint16_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<uint32_t>::min() == double_to_int<uint32_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<uint64_t>::min() == double_to_int<uint64_t>(x, ceil_func));
+
+    x = std::numeric_limits<double>::infinity();
+
+    EXPECT_TRUE(std::numeric_limits<int8_t>::max() == double_to_int<int8_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<int16_t>::max() == double_to_int<int16_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<int32_t>::max() == double_to_int<int32_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<int64_t>::max() == double_to_int<int64_t>(x, ceil_func));
+
+    EXPECT_TRUE(std::numeric_limits<uint8_t>::max() == double_to_int<uint8_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<uint16_t>::max() == double_to_int<uint16_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<uint32_t>::max() == double_to_int<uint32_t>(x, ceil_func));
+    EXPECT_TRUE(std::numeric_limits<uint64_t>::max() == double_to_int<uint64_t>(x, ceil_func));
+
+    auto floor_func = [](double x) { return std::floor(x); };
+
+    x = -std::numeric_limits<double>::infinity();
+
+    EXPECT_TRUE(std::numeric_limits<int8_t>::min() == double_to_int<int8_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<int16_t>::min() == double_to_int<int16_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<int32_t>::min() == double_to_int<int32_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<int64_t>::min() == double_to_int<int64_t>(x, floor_func));
+
+    EXPECT_TRUE(std::numeric_limits<uint8_t>::min() == double_to_int<uint8_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<uint16_t>::min() == double_to_int<uint16_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<uint32_t>::min() == double_to_int<uint32_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<uint64_t>::min() == double_to_int<uint64_t>(x, floor_func));
+
+    x = std::numeric_limits<double>::infinity();
+
+    EXPECT_TRUE(std::numeric_limits<int8_t>::max() == double_to_int<int8_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<int16_t>::max() == double_to_int<int16_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<int32_t>::max() == double_to_int<int32_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<int64_t>::max() == double_to_int<int64_t>(x, floor_func));
+
+    EXPECT_TRUE(std::numeric_limits<uint8_t>::max() == double_to_int<uint8_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<uint16_t>::max() == double_to_int<uint16_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<uint32_t>::max() == double_to_int<uint32_t>(x, floor_func));
+    EXPECT_TRUE(std::numeric_limits<uint64_t>::max() == double_to_int<uint64_t>(x, floor_func));
 }
 
```

Comment:
We should add a LIT test for unsigned comparison to make sure we have it covered as we have other ops and types.

Reply:
Done.